library/core/src/num/nonzero.rs

```diff
@@ -2265,3 +2265,89 @@ mod verify {
     nonzero_check!(u128, core::num::NonZeroU128, nonzero_check_new_unchecked_for_u128);
     nonzero_check!(usize, core::num::NonZeroUsize, nonzero_check_new_unchecked_for_usize);
 }
+
+#[cfg(kani)]
+mod macro_nonzero_check_max {
+    use super::*;
+    macro_rules! nonzero_check_max {
+        ($t:ty, $nonzero_type:ty, $nonzero_check_max_for:ident) => {
+            #[kani::proof]
+            pub fn $nonzero_check_max_for() {
+                let x: $t = kani::any();
+                kani::assume(x != 0);
+                let y: $t = kani::any();
+                kani::assume(y != 0);
+            
+                unsafe {
+                    let x = <$nonzero_type>::new_unchecked(x);
+                    let y = <$nonzero_type>::new_unchecked(y);
+                }
```

Comment:
Please don't re-use names. See also comments on other PRs.

Reply:
Have resolved this problem